content/en/platform/corda/4.12/community/release-notes.md

```diff
@@ -18,233 +18,215 @@ tags:
 
 ---
 
-# Corda Community Edition 4.11 release notes
+# Corda Open Source Edition 4.12 release notes
 
-## Corda Community Edition 4.11.1 release notes
+The Corda Open Source Edition 4.12 release introduces upgrades to the Java and Kotlin versions, along with associated upgrade support. Apart from the features supporting the Java and Kotlin upgrade, no other major new features have been introduced. In this release Java has been upgraded to Java 17 from Java 8 and Kotlin has been upgraded to Kotlin 1.9.20 from 1.2.71.
 
-Corda Community Edition 4.11.1 is a patch release of Corda Community Edition focused on resolving issues.
+When a CorDapp(s) and a node are successfully upgraded to 4.12, you are able to seamlessly interoperate 4.12 and 4.11 (or earlier) nodes on the same network, including the existing transactions on the ledger.
 
-### Fixed issues
+Supporting new Java and Kotlin versions is a major feature, as we must also handle legacy contracts from existing backchains. The upgraded Java and Kotlin versions also have implications for CorDapp developers. Simply replacing the Corda JAR without introducing other changes is not possible.
 
-* Interoperability fix between 4.11 and pre-4.11 nodes when sending/fetching transactions for new data type: `TRANSACTION_RECOVERY`.
+## Upgrade recommendation
 
-## Corda Community Edition 4.11 release notes
+As a developer or node operator, you should upgrade to the [latest released version of Corda]({{< relref "../community/_index.md" >}}) as soon as possible. The latest Corda Open Source release notes are on this page, and for the latest upgrade guide, refer to [Corda Open Source Edition 4.11 to 4.12 upgrade guide]({{< relref "comm-upgrade-guide.md" >}}).
 
-Corda Community Edition 4.11 includes several new features, enhancements, and fixes.
+The steps from this guide only work for direct upgrades from Corda 4.11 to 4.12. If you have any nodes on versions 4.10 and below, you must upgrade them to 4.11 first. To do that, consult the relevant release upgrade documentation.
 
 ## Platform version change
 
-Corda 4.11 uses platform version 13.
+Corda 4.12 uses platform version 140.
 
 For more information about platform versions, see [Versioning]({{< relref "versioning.md" >}}).
 
-## New features and enhancements
+## New features, enhancements and restrictions
+
+### Java and Kotlin upgrade
+
+Corda 4.12 requires Java 17 and Kotlin 1.9.20. This means that you must recompile any legacy CorDapps written for 4.11 or earlier to work with Java 17 and Kotlin 1.9 to be compatible with Corda 4.12. These upgrades enhance the supportability and security of Corda.
+
+### Java 17 compatible releases of Corda SDKs
+
+A number of SDKs and libraries are provided with the base Corda package. These are their Java 17 and Kotlin 1.9 versions:
```

Comment:
```suggestion
A number of SDKs and libraries are provided with the base Corda package. These are the Corda internal versions compatible with Java 17:
```

Not sure the original was correct, maybe this is?

Reply:
Yes, that might not be very clear. Rephrased.